third_party/nvidia/lib/TritonNVIDIAGPUToLLVM/LoadStoreOpToLLVM.cpp

```diff
@@ -696,6 +699,45 @@ struct AtomicRMWOpConversion
            (elementType.isF16() || elementType.isBF16() || elementType.isF32());
   }
 
+  bool isPromotableToNVPTXLD(triton::AtomicRMWOp op) const {
+    if (disableLDAcquireLowering)
+      return false;
+
+    Type valueTy =
+        getTypeConverter()->convertType(getElementTypeOrSelf(op.getType()));
+
+    if (!valueTy.isIntOrFloat() || valueTy.getIntOrFloatBitWidth() != 32)
+      return false;
```

Comment:
Why limit this to 32 bits?

Reply:
An older version of this PR had some PTXAS issues where I wasnt sure if the ld.acquire/relaxed supported the types but all the 16 and 64 bit types seem to work fine with this one. I allow them now.